Doc/whatsnew/3.13.rst

```diff
@@ -172,6 +172,297 @@ Deprecated
+  your code *requires* ``'fork'``.  See :ref:`multiprocessing-start-methods`.
+
+* :mod:`pathlib`: :meth:`~pathlib.PurePath.is_relative_to`,
+  :meth:`~pathlib.PurePath.relative_to`: passing additional arguments is
+  deprecated.
+
+* :func:`pkgutil.find_loader` and :func:`pkgutil.get_loader`
+  now raise :exc:`DeprecationWarning`;
+  use :func:`importlib.util.find_spec` instead.
+  (Contributed by Nikita Sobolev in :gh:`97850`.)
+
+* :mod:`pty`:
+
+  * ``master_open()``: use :func:`pty.openpty`.
+  * ``slave_open()``: use :func:`pty.openpty`.
+
+* :func:`shutil.rmtree` *onerror* parameter is deprecated in 3.12,
+  and will be removed in 3.14: use the *onexc* parameter instead.
+
+* :mod:`sqlite3`:
+
+  * :data:`~sqlite3.version` and :data:`~sqlite3.version_info`.
+
+  * :meth:`~sqlite3.Cursor.execute` and :meth:`~sqlite3.Cursor.executemany`
+    if :ref:`named placeholders <sqlite3-placeholders>` are used and
+    *parameters* is a sequence instead of a :class:`dict`.
+
+  * date and datetime adapter, date and timestamp converter:
+    see the :mod:`sqlite3` documentation for suggested replacement recipes.
+
+* :class:`types.CodeType`: Accessing ``co_lnotab`` was deprecated in :pep:`626`
+  since 3.10 and was planned to be removed in 3.12,
+  but it only got a proper :exc:`DeprecationWarning` in 3.12.
+  May be removed in 3.14.
+  (Contributed by Nikita Sobolev in :gh:`101866`.)
+
+* :mod:`typing`: :class:`~typing.ByteString`, deprecated since Python 3.9,
+  now causes a :exc:`DeprecationWarning` to be emitted when it is used.
+
+* :class:`!urllib.parse.Quoter`.
+
+* :mod:`xml.etree.ElementTree`: Testing the truth value of an
+  :class:`~xml.etree.ElementTree.Element` is deprecated and will raise an
+  exception in Python 3.14.
+
+Pending Removal in Python 3.15
+------------------------------
+
+* :class:`typing.NamedTuple`:
+
+  * The undocumented keyword argument syntax for creating NamedTuple classes
+    (``NT = NamedTuple("NT", x=int)``) is deprecated, and will be disallowed in
+    3.15. Use the class-based syntax or the functional syntax instead.
+
+  * When using the functional syntax to create a NamedTuple class, failing to
+    pass a value to the 'fields' parameter (``NT = NamedTuple("NT")``) is
+    deprecated. Passing ``None`` to the 'fields' parameter
+    (``NT = NamedTuple("NT", None)``) is also deprecated. Both will be
+    disallowed in Python 3.15. To create a NamedTuple class with 0 fields, use
+    ``class NT(NamedTuple): pass`` or ``NT = NamedTuple("NT", [])``.
+
+* :class:`typing.TypedDict`: When using the functional syntax to create a
+  TypedDict class, failing to pass a value to the 'fields' parameter (``TD =
+  TypedDict("TD")``) is deprecated. Passing ``None`` to the 'fields' parameter
+  (``TD = TypedDict("TD", None)``) is also deprecated. Both will be disallowed
+  in Python 3.15. To create a TypedDict class with 0 fields, use ``class
+  TD(TypedDict): pass`` or ``TD = TypedDict("TD", {})``.
```

Comment:
Hmm, these two are listed in the main "Deprecated" section.

Should they be in both or just one?

Looking at older What's New pages, some (but not all) sections have intro texts saying they're were only for deprecations from _earlier_ versions:

* What’s New In Python 3.11
  * Deprecated
    * "This section lists Python APIs that have been deprecated in Python 3.11."
    * Pending Removal in Python 3.12
      * "The following Python APIs have been deprecated in earlier Python releases, and will be removed in Python 3.12."
* What’s New In Python 3.12
  * Deprecated
    * [no intro text]
    * Pending Removal in Python 3.13
      * "The following modules and APIs have been deprecated in earlier Python releases, and will be removed in Python 3.13."
    * Pending Removal in Python 3.14
      * [no intro text]
    * Pending Removal in Future Versions
      * "The following APIs were deprecated in earlier Python versions and will be removed, although there is currently no date scheduled for their removal."


re:

* https://docs.python.org/dev/whatsnew/3.11.html#deprecated
* https://docs.python.org/dev/whatsnew/3.12.html#deprecated